### ISSUE TITLE
Data the score is calculated on differs based on whether the package is installed locally or not

### ISSUE THREAD

**Issue by timtreis** [pharmaR/riskmetric#206]
I tried to automatically scrape all of CRAN for riskmetric scores and quickly noticed that the results massively favoured locally installed packages. After some more digging, it seems that, when pulling from CRAN instead of local, many of the metrics fail.  

Is this behaviour local to my machine?

![image](https://user-images.githubusercontent.com/12818242/124266295-1233c900-db37-11eb-96e9-47e438f293e3.png)
![image](https://user-images.githubusercontent.com/12818242/124266554-6ccd2500-db37-11eb-8605-1bbe1ab02bd6.png)

**Comment by dgkf** [comment]
Thanks @timtreis - there are a few layers to this. 

First, it looks like there's a bug here, because some of those absolutely should not be different between where they're installed. The `license` risk being different is particularly surprising, as the calculation for this is quite straightforward:

https://github.com/pharmaR/riskmetric/blob/1ab25d84d8c3421a623b46e78e802383486de790/R/assess_license.R#L28-L31

Bugs aside, it _is_ expected that some values differ depending on environment. For example, even on CRAN, some packages report different output of `R CMD check` on different platforms (OSX, windows, linux, Solaris). This is why we contextualize risk by info source in the first place - because package behaviors can and do change depending on host system, internet access, available system libraries or whether suggested packages are installed.

These can manifest as different metrics even. For example, `R CMD check` on the host system might be one measure of risk locally, but `R CMD check` running successfully across a range of platforms might also be a measure of the transportability of a package.

**Comment by timtreis** [comment]
Hey Doug, thanks for getting back to me! The R_CMD_Check is at least consistent in the sense that it fails in both cases :D I did not do a deep-dive on the `riskmetric` codebase yet, but it seems that (given internet connectivity) a lot of the information could be scraped from the automatically generated CRAN pages more reliably? 

For example the "news_current", "has_vignettes", "has_bug_reports_url", "has_maintainer", "has_news", "has_source_control" seem to be extractable from pages like https://cran.r-project.org/package=visR.

Is there a benefit of choosing the as.environment(...) approach over webscraping for the CRAN usecase?

**Comment by dgkf** [comment]
@timtreis A lot of the information is scraped from the CRAN page already. For example, here is how the maintainers are provided when a package info is sourced from a `pkg_remote` (CRAN or Bioconductor) object:

https://github.com/pharmaR/riskmetric/blob/1ab25d84d8c3421a623b46e78e802383486de790/R/pkg_ref_cache_maintainer.R#L13-L17

The goal is to use CRAN's material when possible if a package isn't installed. Each relevant page is accessed only once and parsed out into different fields. The issue you're experiencing looks like it's a bug in the processing. 

--- 

On a side note, I did some digging and found a regression in a recent PR that is the root of the issues you identified. I'm going to file a separate issue to address this with more details about the internal bug. I will close this issue once that's been filed, but feel free to continue reaching out if you have questions about the internals of the package :+1:

**Comment by dgkf** [comment]
Alright - an isolated bug report was filed in #207 and I issued a PR with a fix. If you'd like, you're welcome to try out the bugfix branch

```r
remotes::install_github("pharmaR/riskmetric@207-pkg-ref-hierarchy")

library(dplyr)
library(riskmetric)

# visR not installed 
as_tibble(pkg_ref("visR")) %>%
  pkg_assess() %>%
  pkg_score()
```

There might still be some discrepancies, as not all metrics are implemented for scraping off of CRAN pages, but the results now should be much closer to what you see when the package is installed. 

Many thanks for your bug report - it highlighted an oversight in one of our recent refactors.

**Comment by timtreis** [comment]
Hey Doug, thanks for the quick fix! I tried it again and now the installation status doesn't matter any more. Fantastic!

![image](https://user-images.githubusercontent.com/12818242/124722897-c7d19400-df0a-11eb-8595-5d8fad5b3baa.png)
